File: 0204-count-primes/0204-count-primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
  const prem = new Array(n).fill(0);
  let ans = 0;
  
  for (let i = 2; i < n; i++) {
    if (prem[i]) continue;
    
    ans++;
    for (let j = i * i; j < n; j += i) prem[j] = 1;
  }
  
  return ans;
};

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        prem = [0] * n
        ans = 0
        for i in range(2, n):
            if prem[i] == 1: continue
            
            ans += 1
            for j in range(i*i, n, i):
                prem[j] = 1
        return ans